### ISSUE TITLE
Create an issue report page on the website and automatically close issue reports on GitHub

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#93485]
vuejs created their own issue reporter that opens github with a pre-filled template and closes via a boy all other issues:

Here's what their new issue page looks like https://github.com/vuejs/vue/issues/new

![image](https://user-images.githubusercontent.com/2193314/77647512-c8b40180-6f23-11ea-8a31-2fca805f2b54.png)

It directs you to https://new-issue.vuejs.org/ which gives a form with required fields and then uses `window.open` to set the title, body and label of the issue:

https://github.com/vuejs/vue-issue-helper/blob/4741cb04766bf7e9b6fa632f040e0c6c82dd5b36/src/components/App.vue#L220-L223

They have a nice explainer why they do this on the issue reporter:

![image](https://user-images.githubusercontent.com/2193314/77647916-83440400-6f24-11ea-9edd-907a5eb4158c.png)

If we implemented something similar (`code.visualstudio.com/issue`?) then it would solve several problems that bots have difficulty with like automatic labelling (tried and failed via bots), requiring version info (https://github.com/microsoft/vscode/issues/92958) and requiring a more strict template in general. This would increase overall issue quality, reducing the number of iterations issues need to go through and could even help guide the user to self-help documentation in the future which would be a great way to cut down on the volume of issues we receive. For example consider the following flow:

- Open issue reporter
- Select bug category
- Select terminal component (this would result in the integrated-terminal label and automatically triage to me)
- Since terminal was selected, a link to https://github.com/microsoft/vscode/wiki/Terminal-Issues is presented (or the same information is inlined) which would either result in the user not reporting the issue or in a better quality issue as I often link this page https://github.com/microsoft/vscode/wiki/Terminal-Issues
- Enter VS Code version (OS could be automatically prefilled using a best effort detection from the user agent)
- Submit

Yes it's a little more overhead but it would help us greatly. Additionally we could localize everything on the issue reporter itself so if we could encourage English but give people other options and let them fill in the form in their own language which is often more useful then the user computer translating them on their side.

Issues created by our team would not be auto closed obviously.

**Comment by jrieken** [comment]
I think this is the worst you can do to the community and to GH itself

**Comment by JacksonKearl** [comment]
Would this have the "URL is too long please paste" issue?

**Comment by Tyriar** [comment]
@JacksonKearl it could hit that as we would not ask users to auth with GH, but I think the main reason we have that problem with the built-in reporter is because of the extensions list.

**Comment by ultraGentle** [comment]
Related: #89652

**Comment by jrieken** [comment]
>  but I think the main reason we have that problem with the built-in reporter is because of the extensions list.

Well, we are adding APIs for authentication provider and there will be GH auth provider builtin. So, when user file an issue that's too large we simply ask them to log in.

**Comment by ultraGentle** [comment]
1.  I think it would be a good idea to expose the issue-reporting functionality within the editor, even just as a command that takes you to the web portal, so that users don't have to "figure out" where to go to report an error.  Even the current GitHub system preferences those who are familiar with the open source community and procedures.  An in-editor approach would democratize.  

2. There's a lot of inefficiency in folks reporting duplicate or related issues, so if the issue-reporter could tag/aggregate existing issues by category (highlighting, navigation, debugging etc.), it would be easier than at present for folks to jump on board an existing issue.  

3.  How would this really affect the feature request flow?  If folks are submitting issues through a custom portal, then they are less likely to see and vote on FRs here.  It's already hard to get enough attention on an issue to get the requisite votes, and the voting seems already skewered toward power users who spend a lot of time actively in VSC's GitHub.  This might exacerbate that unless (as in 2, above) the portal also aggregated feature requests, allowing for easy perusal and voting of FRs.

**Comment by JacksonKearl** [comment]
@ultraGentle For 1, we do already have the Report Issue command, and the status bar item for feedback
![image](https://user-images.githubusercontent.com/8586769/77698602-22411e00-6f6e-11ea-83b0-ddc3b91881a3.png)

**Comment by ultraGentle** [comment]
@JacksonKearl Thanks for the heads up!  Long time user, but never knew you could do that.  

But isn't that widget part of the problem (with regard to the issue raised in this post)?  

1.  Tweeting feedback is even less structured than the current GitHub system, and 

2.  There's no way from that widget to see whether your issue already exists, encouraging duplicate submissions, which take time to write and read.   True, it links to GitHub, but then you're back to the beginning of this thread.  

Thumbs upping/downing existing issues is more efficient, which is why I was suggesting aggregating issues and FRs in the earlier post. 

(Apologies for the numbered lists -- just helps me organize my thoughts.)

**Comment by JacksonKearl** [comment]
It actually opens this view, which is good because from here the issue created is profiled with version information/etc. that we need.
![image](https://user-images.githubusercontent.com/8586769/77701895-81a22c80-6f74-11ea-8eaa-e80bc8a4b45f.png)

I thought that it would additionally show similar existing issues, but that seems not to be the case.

**Comment by ultraGentle** [comment]
That's great!  Maybe it's kind of buried, though?  If I saw a big "tweet this!" Or even a "how was your experience?" I'd probably close it, since big data companies have been so aggressive at soliciting feedback lately that it's off-putting --that's what I always thought that little smiley in the corner was.  Maybe worth separating concerns there? 

One thing that might help more concretely is whatever the Google Firebase team has going on their GitHub: they have a required template for issues, and if you don't follow it, you can't create the issue. I'm not sure what the mechanism is, but it happens from within their repository, not from an external website. But it's of the same ilk as the idea in the original post in this thread.

**Comment by JacksonKearl** [comment]
We are actually in the process of making such a bot. It's currently live, but under active discussion:
https://github.com/microsoft/vscode/issues/92958

**Comment by ultraGentle** [comment]
Nifty.  If the bot could auto-tag bugs/features by category, and those tags were visible at the top of the issues, you might get more discussion on fewer threads, which would make it easier for  everyone to follow.  

As someone with no vested interest except for a love of text editors, there's such a high volume of new issues that it's hard to follow, and therefore to contribute (so I can only imagine what it's like for you guys!).

Anyway, I'll stop with the aggregation plugs.  You guys make an awesome product!

**Comment by Tyriar** [comment]
> one thing that might help is whatever the Google Firebase team has going on their GitHub: they have a required template for issues, and if you don't follow it, you can't create the issue (iirc, or it may be immediately flagged by a bot).

Yes, considering the reaction from the team internally about having the user leave GH to report an issue, that's one way forward that might work. Going to close this out as I don't see it happening 😄
